cfme/tests/cloud/test_providers.py

```diff
@@ -1245,12 +1257,22 @@ def test_ec2_create_sns_topic():
             1.
             2. Check SNS topic for this region in AWS
     """
-    pass
+    request.addfinalizer(lambda: provider.delete_if_exists())
```

Comment:
Can you include an in-line comment please explaining why you're explicitly calling this.